staff/staff_views_helper.py

```diff
@@ -27,70 +26,117 @@ def create_user_with_profile(form, random_password):
+    create_classroom_membership(teacher_assistant, classroom, "teacher_assistant")
     enroll_in_class(form, form.cleaned_data.get("teacher"))
     enroll_in_class(form, form.cleaned_data.get("teacher_assistant"))
-    email_classroom(request, teacher_email_list, classroom.course)
-    return classroom
-
-
-def add_volunteers_and_students_to_classroom(request, form, classroom):
-    email_list = []
     for volunteer in form.cleaned_data.get("volunteers"):
         enroll_in_class(form, volunteer)
-        django_user = UserProfile.objects.get(
-            salesforce_id=volunteer.client_id.lower()
-        ).user_id
-        classroom.volunteers.add(django_user)
-        email_list.append(DjangoUser.objects.get(id=django_user).email)
+        django_volunteer = get_django_user_from_contact(volunteer)
+        create_classroom_membership(django_volunteer, classroom, "volunteer")
+        email_list.append(django_volunteer.email)
     for student in form.cleaned_data.get("students"):
         enroll_in_class(form, student)
-        django_user = UserProfile.objects.get(
-            salesforce_id=student.client_id.lower()
-        ).user_id
-        classroom.students.add(django_user)
-        email_list.append(DjangoUser.objects.get(id=django_user).email)
+        django_student = get_django_user_from_contact(student)
+        create_classroom_membership(django_student, classroom, "student")
+        email_list.append(django_student.email)
+    generate_classroom_sessions_and_attendance(classroom)
     email_classroom(request, email_list, classroom.course)
 
 
+def get_django_user_from_contact(contact):
+    return DjangoUser.objects.get(id=UserProfile.objects.get(salesforce_id=contact.client_id.lower()).user_id)
+
+
+def retrieve_userprofile_from_form(form, name_string):
+    return UserProfile.objects.get(
+            salesforce_id=form.cleaned_data.get(name_string).client_id.lower()
+        )
+
+
+def create_classroom_membership(django_user_member, classroom, membership_type):
+    cm = ClassroomMembership(
+        member=django_user_member,
+        classroom=classroom,
+        membership_type=membership_type
+    )
+    cm.save()
+
+
+def generate_classroom_sessions_and_attendance(classroom):
+    classroom.attendance_summary = {
+        "attendance_statistic": get_course_attendance_statistic(classroom.id)
+    }
+    classroom.save()
+    class_offering = ClassOffering.objects.get(name=classroom.course)
+    dates = class_offering_meeting_dates(class_offering)
+    sessions = [Session(
+        classroom_id=classroom.id,
+        date=day
+    ) for day in dates]
+    Session.objects.bulk_create(sessions)
+    classroom_students = get_users_of_type_from_classroom(classroom, "student")
+    attendances = [Attendance(
+            student_id=student.id,
+            session_id=sessions[x].id,
+            classroom_id=classroom.id,
+            date=day,
+        ) for student in classroom_students for x, day in enumerate(dates)]
```

Comment:
This is a use case for [zip](https://docs.python.org/3/library/functions.html#zip)

```python
        attendances = [Attendance(
            student_id=student.id,
            session_id=session.id,
            classroom_id=classroom.id,
            date=day,
        ) for student in classroom_students for session, day in zip(sessions, dates)]
```

Reply:
Thank you, I will use it

Reply:
After a second look you don't really even need the zip, because you have `session.date` which could be used instead of `day`.

Reply:
Ok

Reply:
Fixed in 5688a46e57d2ae589bb11041271b08866d2cdf21